Lesson 2 - Functions/code/homework/code.js

```diff
@@ -23,23 +23,47 @@
     //////////////////////////////////////////////////////////////////////////////////////
 
     global.UAM.addAircraft = function (newAircraftCode) {
-        // function should return new aircraft object
+        var length = global.UAM.aircrafts.push({code: newAircraftCode, services: []});
+        
+        return global.UAM.aircrafts[length - 1];
     };
 
     global.UAM.removeAircraft = function (aircraftObj) {
-        // !!!
+        var pos = global.UAM.aircrafts.indexOf(aircraftObj);
+        
+        if(pos >= 0) {
+            global.UAM.aircrafts.splice(pos, 1);
+        }
     };
 
     global.UAM.addWorkToAircraft = function(aircraftObj, name, timeToExxecute) {
-        // !!!
+        aircraftObj.services.push({name: name, timeToExecute: timeToExxecute});
     };
         
-    global.UAM.reduceTimeToExecute = function(time) {
-        // !!!
+    global.UAM.reduceTimeToExecute = function(aircraftObj, time) {
+        if(aircraftObj.services !== null) {
+            aircraftObj.services.forEach(function(elem) {
+                elem.timeToExecute -= time;
+            });
+        }
+        
     };
     
     global.UAM.getAircraftsForRepairs = function(maxTimeToExecute) {
-        // !!!
+        var airForRep = [];
+        
+        global.UAM.aircrafts.forEach(function(elem) {
+            if(elem.services !== null) {
+                for(var i = 0; i < elem.services.length; i++) {
+                    if(elem.services[i].timeToExecute <= maxTimeToExecute) {
+                        airForRep.push(elem);
+                        break;
```

Comment:
Something like that. 

```javascript
var airForRep = [];
        global.UAM.aircrafts.forEach(function (elem) {
            if (elem.services) {
                elem.services.find(function (item) {
                    if (item.timeToExecute <= maxTimeToExecute) {
                        airForRep.push(elem);
                        return elem;
                    }
                });
            }
        });

        return airForRep;
```

But ok, this could to overkill for this kind of task.